src/wasm_split_tools/mod.rs

```diff
@@ -106,37 +110,50 @@ function makeLoad(url, deps) {
 "#,
     );
     let mut split_deps = HashMap::<String, Vec<String>>::new();
-    for (name, _) in split_program_info.output_modules.iter() {
-        let SplitModuleIdentifier::Chunk { splits, hash } = name else {
+    for (identifier, _) in split_program_info.output_modules.iter() {
+        let SplitModuleIdentifier::Chunk { splits, .. } = identifier else {
             continue;
         };
-        for split in splits {
+        let chunk_name_hashed = identifier.name_hashed(proj);
+        let chunk_name_sanitized = chunk_name_hashed.replace(['.', '-'], "_");
+
+        for split_name in splits {
             split_deps
-                .entry(split.clone())
+                .entry(split_name.clone())
                 .or_default()
-                .push(name.name_hashed(proj));
+                .push(chunk_name_sanitized.clone());
+        }
+
+        if !chunk_name_hashed.is_empty() {
+            javascript.push_str(
+                format!(
+                    "const __wasm_split_load_{chunk_name_sanitized} = makeLoad(new URL(\"./{chunk_name_hashed}.wasm\", import.meta.url), []);\n",
+                ).as_str()
+            )
         }
-        javascript.push_str(format!(
-            "const __wasm_split_load_{name} = makeLoad(new URL(\"./{name}.{hash}.wasm\", import.meta.url), []);\n",
-            name = name.name(proj),
-        ).as_str())
     }
+
     for (identifier, _) in split_program_info.output_modules.iter().rev() {
-        if matches!(identifier, SplitModuleIdentifier::Chunk { .. }) {
+        if !matches!(identifier, SplitModuleIdentifier::Split { .. }) {
             continue;
         }
-        let name = identifier.name(proj);
         let name_hashed = identifier.name_hashed(proj);
-        javascript.push_str(format!(
-            "export const __wasm_split_load_{name} = makeLoad(new URL(\"./{name_hashed}.wasm\", import.meta.url), [{deps}]);\n",
-            deps = split_deps
-            .remove(&name_hashed)
-            .unwrap_or_default()
-            .iter()
-            .map(|x| format!("__wasm_split_load_{x}"))
-            .collect::<Vec<_>>()
-            .join(", "),
-        ).as_str())
+        if !name_hashed.is_empty() {
+            if let SplitModuleIdentifier::Split { name, .. } = identifier {
+                javascript.push_str(
+                    format!(
+                        "export const __wasm_split_load_{name} = makeLoad(new URL(\"./{name_hashed}.wasm\", import.meta.url), [{deps}]);\n",
+                        deps = split_deps
+                            .remove(name)
+                            .unwrap_or_default()
+                            .iter()
+                            .map(|x| format!("__wasm_split_load_{x}"))
+                            .collect::<Vec<_>>()
+                            .join(", "),
+                    ).as_str()
+                )
+            }
+        }
```

Comment:
you could get rid of one level of nesting like this, also deduplicates checking the identifier

```suggestion
        let SplitModuleIdentifier::Split { name, .. } = identifier else {
            continue;
        };
        let name_hashed = identifier.name_hashed(proj);
        if !name_hashed.is_empty() {
            javascript.push_str(
                format!(
                    "export const __wasm_split_load_{name} = makeLoad(new URL(\"./{name_hashed}.wasm\", import.meta.url), [{deps}]);\n",
                    deps = split_deps
                        .remove(name)
                        .unwrap_or_default()
                        .iter()
                        .map(|x| format!("__wasm_split_load_{x}"))
                        .collect::<Vec<_>>()
                        .join(", "),
                ).as_str()
            )
        }
```

Reply:
Great catch, thank you! I'll definitely use that let-else refactor.

My PR was auto-closed because the base branch was removed, but I'll open a new one soon and will include your suggestion. Thanks!